src/ExperimentConfigurationRequester.php

```diff
@@ -49,11 +49,11 @@ public function getConfiguration(string $experiment): ?ExperimentConfiguration
 
         if (!$configuration) {
             $configurations = $this->fetchAndStoreConfigurations();
-            if (!$configurations || !count($configurations)) {
+            if (!$configurations || !count($configurations) || !array_key_exists($experiment, $configurations)) {
                 return null;
             }
 
-            $configuration = $this->fetchAndStoreConfigurations()[$experiment];
+            $configuration = $configurations[$experiment];
```

Comment:
I need to understand this a bit more 

`ExperimentConfiguration` expects an `array` - does php treat everything as an array or are there objects too? In our Java implementation we pass the entire `$configurations` into the `ExperimentConfiguration` object which is a:

```
private Map<String, ExperimentConfiguration> flags;
```

Ok I buy it, in PHP the `ExperimentConfiguration` is the singular type so we need to look it up.

Reply:
PHP does have objects but they are not at play here (and not often used in situations like these)